aten/src/ATen/native/cuda/BinaryOpsKernel.cu

```diff
@@ -13,12 +13,19 @@
 namespace at { namespace native {
 
 void add_kernel_cuda(TensorIterator& iter, Scalar alpha_scalar) {
-  AT_DISPATCH_ALL_TYPES_AND(kHalf, iter.dtype(), "add_cuda", [&]() {
-    auto alpha = alpha_scalar.to<scalar_t>();
-    gpu_kernel_with_scalars(iter, [alpha]GPU_LAMBDA(scalar_t a, scalar_t b) -> scalar_t {
-      return a + alpha * b;
+  if (iter.dtype() == ScalarType::Bool) {
```

Comment:
This looks the same as the non-bool case. Maybe combine them with AT_DISPATCH_ALL_TYPES_AND2?